### ISSUE TITLE
ec2 models get_zone_by_name KeyError when creating a Batch client

### ISSUE THREAD

**Issue by bhayden53** [spulec/moto#4687]
A pytest-based github action that depends on moto has started failing due to a traceback in ec2 models. The version of moto installed was:

    Collecting moto[all]
    Downloading moto-2.2.18-py2.py3-none-any.whl (1.5 MB)

The full traceback was:
```==================================== ERRORS ====================================
______________________ ERROR at setup of test_blackboard _______________________

aws_credentials = None

    @pytest.fixture(scope="function")
    def batch_client(aws_credentials):
>       with mock_batch():

tests/conftest.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/__init__.py:17: in f
    module = importlib.import_module(module_name, "moto")
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/batch/__init__.py:1: in <module>
    from .models import batch_backends
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/batch/models.py:14: in <module>
    from moto.ec2 import ec2_backends
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/__init__.py:1: in <module>
    from .models import ec2_backends
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:8729: in <module>
    region.name: EC2Backend(region.name) for region in RegionsAndZonesBackend.regions
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:8729: in <dictcomp>
    region.name: EC2Backend(region.name) for region in RegionsAndZonesBackend.regions
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:8624: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1141: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1374: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1390: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1513: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:3470: in __init__
    super().__init__()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1743: in __init__
    self._load_amis()
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1752: in _load_amis
    self.amis[ami_id] = Ami(self, **ami)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:1701: in __init__
    volume = self.ec2_backend.create_volume(size=15, zone_name=region_name)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:3486: in create_volume
    zone = self.get_zone_by_name(zone_name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <moto.ec2.models.EC2Backend object at 0x7f5d679f9710>
name = 'us-east-1a'

    def get_zone_by_name(self, name):
>       for zone in self.zones[self.region_name]:
E       KeyError: 'ap-southeast-3'

/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/moto/ec2/models.py:2209: KeyError
```
The github action in question is here:
https://github.com/spacetelescope/calcloud/runs/4512515777?check_suite_focus=true

And the code that's being executed when it fails seems to be this:
```
@pytest.fixture(scope="function")
def batch_client(aws_credentials):
    with mock_batch():
        yield boto3.client("batch", region_name="us-east-1")
```

The last run one week ago with the same source code worked fine, using moto 2.2.17
```
Collecting moto[all]
  Downloading moto-2.2.17-py2.py3-none-any.whl (1.5 MB)
```
https://github.com/spacetelescope/calcloud/runs/4434257100?check_suite_focus=true

**Comment by YustinS** [comment]
Can confirm the same issue encountered using mock_ssm.

In my situation found it appears to arise from a botocore release, if I pinned version `1.23.23` (instead of `1.23.24`) and boto3 to `1.20.23` (just for consistency since botocore and boto3 are tied together).

Not sure if that's overly helpful, but I'll mention it as a possible interim workaround.

**Comment by foxmulder900** [comment]
Presumably an entry for `ap-southeast-3` needs added here:

https://github.com/spulec/moto/blob/master/moto/ec2/models.py#L2096

But I'm not sure exactly what all the values should be.

**Comment by bhayden53** [comment]
In my last good run I had
```
Collecting botocore<1.24.0,>=1.23.20
  Downloading botocore-1.23.20-py3-none-any.whl (8.4 MB)
```
And in the failed run I had
```
Collecting botocore<1.24.0,>=1.23.24
  Downloading botocore-1.23.24-py3-none-any.whl (8.4 MB)
```

So I pinned botocore to 1.23.20, and that run was successful against moto 2.2.18
```
...
Collecting moto[all]
  Downloading moto-2.2.18-py2.py3-none-any.whl (1.5 MB)
...
Collecting botocore==1.23.20
  Downloading botocore-1.23.20-py3-none-any.whl (8.4 MB)
```
https://github.com/spacetelescope/calcloud/runs/4513019218?check_suite_focus=true

**Comment by pf-billoday** [comment]
https://github.com/boto/botocore/commit/19f83d8d66b972172591585dbcd8d4d0cdb35603

Looks like that's the commit that broke things - unfortunately not able to observe anything WRT to the AZs available or I'd put in a PR to remediate.

**Comment by foxmulder900** [comment]
This change did the trick for me:
https://github.com/spulec/moto/pull/4688

I just deduced what those values should be based on what they are for `ap-southeast-1` & `ap-southeast-2`, so it would be good for someone to verify them a little more closely.

**Comment by Slash7GNR** [comment]
I'm too experiencing some unit tests that fails, any idea when this issue is going to be resolved?

**Comment by yaronbenezra** [comment]
is there any update?

**Comment by bblommers** [comment]
Thanks for raising this everyone - this is now fixed as of Moto >= 2.2.19.

**Comment by andreasch86** [comment]
After fixing failing EC2 tests with the upgrade to `moto==2.2.19`, a new failure has arisen on kinesis:

Stack trace:
```
self = <botocore.client.Firehose object at 0x7f7847db9a10>, operation_name = 'CreateDeliveryStream'
api_params = {'DeliveryStreamName': 'foo', 'DeliveryStreamType': 'DirectPut', 'ExtendedS3DestinationConfiguration': {'BucketARN': '...N': 'arn:aws:iam::123456789012:role/firehose_delivery_role', 'TableName': 'outputTable'}}, 'Prefix': 'myFolder/', ...}}

    def _make_api_call(self, operation_name, api_params):
        operation_model = self._service_model.operation_model(operation_name)
        service_name = self._service_model.service_name
        history_recorder.record('API_CALL', {
            'service': service_name,
            'operation': operation_name,
            'params': api_params,
        })
        if operation_model.deprecated:
            logger.debug('Warning: %s.%s() is deprecated',
                         service_name, operation_name)
        request_context = {
            'client_region': self.meta.region_name,
            'client_config': self.meta.config,
            'has_streaming_input': operation_model.has_streaming_input,
            'auth_type': operation_model.auth_type,
        }
        request_dict = self._convert_to_request_dict(
            api_params, operation_model, context=request_context)

        service_id = self._service_model.service_id.hyphenize()
        handler, event_response = self.meta.events.emit_until_response(
            'before-call.{service_id}.{operation_name}'.format(
                service_id=service_id,
                operation_name=operation_name),
            model=operation_model, params=request_dict,
            request_signer=self._request_signer, context=request_context)

        if event_response is not None:
            http, parsed_response = event_response
        else:
            http, parsed_response = self._make_request(
                operation_model, request_dict, request_context)

        self.meta.events.emit(
            'after-call.{service_id}.{operation_name}'.format(
                service_id=service_id,
                operation_name=operation_name),
            http_response=http, parsed=parsed_response,
            model=operation_model, context=request_context
        )

        if http.status_code >= 300:
            error_code = parsed_response.get("Error", {}).get("Code")
            error_class = self.exceptions.from_code(error_code)
>           raise error_class(parsed_response, operation_name)
E           botocore.exceptions.ClientError: An error occurred (UnrecognizedClientException) when calling the CreateDeliveryStream operation: The security token included in the request is invalid.

/usr/local/lib/python3.7/site-packages/botocore/client.py:719: ClientError
```
Installed versions are:
```
boto3==1.20.34
botocore==1.23.34
moto==2.2.19
```
